apis/supernova/src/handler/data-property-handler.js

```diff
@@ -18,77 +46,205 @@ class DataPropertyHandler {
+   * @property {string=} label
+   * @property {qix.NxMeasure=} defaults
+   */
+
+  /**
+   * Sets the properties for the handler.
+   * @param {object=} properties - The properties object to set.
+   * @description Updates the handler's properties and analysis type flag.
+   * @memberof DataPropertyHandler
+   * @example
+   * handler.setProperties({ metaData: { isAnalysisType: true } });
+   */
   setProperties(properties) {
     this.properties = properties;
     this.isAnalysisType = this.properties?.metaData?.isAnalysisType;
   }
 
+  /**
+   * Sets the global change listeners.
+   * @param {Function[]} arr - Array of listener functions.
+   * @description Assigns global change listeners to the handler.
+   * @memberof DataPropertyHandler
+   * @example
+   * handler.setGlobalChangeListeners([listener1, listener2]);
+   */
   setGlobalChangeListeners(arr) {
     this.globalChangeListeners = arr;
   }
 
+  /**
+   * @param {object=} layout - The layout object to set.
+   * @description Sets the layout for the handler.
+   * @memberof DataPropertyHandler
+   * @example
+   * handler.setLayout(layoutObj);
+   */
   setLayout(layout) {
     this.layout = layout;
   }
 
-  static type() {
+  /**
+   * @throws {Error}
+   * @description Throws an error indicating the method must be overridden.
+   * @memberof DataPropertyHandler
+   * @example
+   * DataPropertyHandler.type(); // Throws error
+   */
+  // eslint-disable-next-line class-methods-use-this
+  type() {
     throw new Error('Must override this method');
   }
 
   // ---------------------------------------
   // ---------------DIMENSION---------------
   // ---------------------------------------
 
-  static getDimensions() {
+  /**
+   * @returns {Array} Empty array.
+   * @description Returns the default dimension array.
+   * @memberof DataPropertyHandler
+   */
+  // eslint-disable-next-line class-methods-use-this
+  getDimensions() {
     return [];
   }
 
-  getDimension(id) {
+  /**
+   * Gets a dimension by id from dimensions or alternative dimensions.
+   * @param {LibraryDimension} libraryDimension
+   * @returns {qix.NxDimension} - The found dimension.
+   * @description Searches for a dimension by id in both main and alternative dimensions.
+   * @memberof DataPropertyHandler
+   * @example
+   * const dim = handler.getDimension({ id: 'dimId' });
+   */
+  getDimension(libraryDimension) {
     const dimensions = this.getDimensions();
     const alternativeDimensions = this.getAlternativeDimensions();
 
-    return findFieldById(dimensions, id) ?? findFieldById(alternativeDimensions, id);
+    return findFieldById(dimensions, libraryDimension.id) ?? findFieldById(alternativeDimensions, libraryDimension.id);
   }
 
-  static getAlternativeDimensions() {
+  /**
+   * Throws an error indicating the method must be implemented in subclasses.
+   * @throws {Error}
+   * @memberof DataPropertyHandler
+   */
+  // eslint-disable-next-line class-methods-use-this
```

Comment:
@Caele I ended up disabling typeError to prevent using `static`
typeError: `Expected 'this' to be used by class method 'addMeasure'`